src/timesyncwait/timesyncwaitd.c

```diff
@@ -0,0 +1,124 @@
+/*
+ * systemd service to wait until kernel realtime clock is synchronized
+ *
+ * Copyright 2018 Peter A. Bigot
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
+ * USA
+ */
+
+#include <errno.h>
+#include <signal.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/timerfd.h>
+#include <sys/timex.h>
+
+#include "sd-event.h"
+
+#include "missing.h"
+#include "time-util.h"
+
+typedef struct ClockState {
+        int fd;                        /* non-negative is descriptor from timerfd_create */
+        int adjtime_state;             /* return value from last adjtimex(2) call */
+        sd_event *event;               /* the systemd event loop object */
+        sd_event_source *event_source; /* non-null is the active io event source */
+} ClockState;
+
+static int io_handler(sd_event_source * s,
+                      int fd,
+                      uint32_t revents,
+                      void *userdata) {
+        ClockState *sp = userdata;
+        int rc = 0;
+        struct itimerspec its = {
+                .it_value.tv_sec = TIME_T_MAX,
+        };
+        struct timex tx = {0};
```

Comment:
btw, we use gcc extensions that allow us to write this as `= {}`, i.e. without the extra 0